libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/ValueRecognizer.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Antlr4.Runtime;
+using Microsoft.Bot.Schema;
+using Microsoft.Recognizers.Text.NumberWithUnit;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Recognizers
+{
+    /// <summary>
+    /// ValueRecognizer - InputRecognizer for mapping message activity .Value payload into intent/entities.
+    /// </summary>
+    /// <remarks>
+    /// This recognizer will map MessageActivity Value payloads into intents and entities.
+    ///     activity.Value.intent => RecognizerResult.Intents.
+    ///     activity.Value.properties => RecognizerResult.Entities.
+    /// </remarks>
+    public class ValueRecognizer : InputRecognizer
```

Comment:
Is this functionality already being used?

Reply:
This was hard coded in (Preview) adaptiveDialog itself, which meant that you could never remove it or control the implementation.  This was pulling it out into a recgonizer

---
In reply to: [360621355](https://github.com/microsoft/botbuilder-dotnet/pull/3174#discussion_r360621355) [](ancestors = 360621355)